pkg/cfn/manager/delete_tasks.go

```diff
@@ -181,3 +184,30 @@ func (c *StackCollection) NewTasksToDeleteIAMServiceAccounts(shouldDelete func(s
 
 	return tasks, nil
 }
+
+// NewTasksToDeleteIdentityRoleARNFromAuthConfigMap defines tasks required to delete identity role ARNs from the auth configmap
+// node groups in the cluster config must already be populated with IAM role ARNs
+func (c *StackCollection) NewTasksToDeleteIdentityRoleARNFromAuthConfigMap(cfgMarkedForDeletion *api.ClusterConfig, clientSet kubernetes.Interface) (*TaskTree, error) {
+	tasks := &TaskTree{Parallel: false}
+
+	for _, n := range cfgMarkedForDeletion.NodeGroups {
+		tasks.Append(&authConfigMapTask{
+			info:      fmt.Sprintf("remove identity role arn %q from auth configmap", n.IAM.InstanceRoleARN),
+			clientSet: clientSet,
+			call: func(clientSet kubernetes.Interface) error {
+				return authconfigmap.RemoveARNIdentity(clientSet, n.IAM.InstanceRoleARN, false)
+			},
+		})
+	}
+	for _, n := range cfgMarkedForDeletion.ManagedNodeGroups {
```

Comment:
Please refactor as there is redundant code between these two blocks.

Reply:
I removed the ManagedNodeGroup loop and put in a comment mentioning the aws api doing the deletion